yaksh/views.py

```diff
@@ -853,34 +854,53 @@ def download_csv(request, questionpaper_id):
 
 
 @login_required
-def grade_user(request, username, questionpaper_id=None):
+def grade_user(request, quiz_id = None, user_id=None, attempt_number = None):
+
     """Present an interface with which we can easily grade a user's papers
     and update all their marks and also give comments for each paper.
     """
     current_user = request.user
     ci = RequestContext(request)
     if not current_user.is_authenticated() or not is_moderator(current_user):
         raise Http404('You are not allowed to view this page!')
-    data = get_user_data(username, questionpaper_id)
-    if request.method == 'POST':
-        papers = data['papers']
-        for paper in papers:
-            for question, answers in paper.get_question_answers().iteritems():
-                marks = float(request.POST.get('q%d_marks' % question.id, 0))
-                last_ans = answers[-1]
-                last_ans.marks = marks
-                last_ans.save()
-            paper.comments = request.POST.get(
-                'comments_%d' % paper.question_paper.id, 'No comments')
-            paper.save()
-
-        context = {'data': data}
-        return my_render_to_response('yaksh/user_data.html', context,
-                                     context_instance=ci)
-    else:
-        context = {'data': data}
-        return my_render_to_response('yaksh/grade_user.html', context,
-                                     context_instance=ci)
+
+    course_details =Course.objects.filter(creator = current_user)
+    context = {"course_details": course_details}
+
+    if quiz_id != None:
+        questionpaper_id = QuestionPaper.objects.filter(quiz_id = quiz_id).values("id")
+
+        answerpaper_details = AnswerPaper.objects.filter(question_paper_id = questionpaper_id)
+        users_id = answerpaper_details.values_list("user_id").distinct()
+        user_details = User.objects.filter(id__in = users_id)
+        context = {"users": user_details, "quiz_id": quiz_id}
+        if user_id !=None:
```

Comment:
Could this be refactored into a method on the model to provide the appropriate attempt when asked along with a test for the model?